web/src/components/AppStoreBanner.tsx

```diff
@@ -0,0 +1,111 @@
+import { EmapsIcon } from 'icons/emapsIcon';
+import { X } from 'lucide-react';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import trackEvent from 'utils/analytics';
+
+import { isAndroid, isIphone } from '../features/weather-layers/wind-layer/util';
+import { Button } from './Button';
+
+const IS_APP_BANNER_DISMISSED = 'isAppBannerDismissed';
+
+export enum AppStoreURLs {
+  APPLE = 'https://apps.apple.com/us/app/electricity-maps/id1224594248',
+  GOOGLE = 'https://play.google.com/store/apps/details?id=com.tmrow.electricitymap&hl=en',
+}
+
+enum TrackEvents {
+  APP_BANNER_CTA_CLICKED = 'App Banner CTA Clicked',
+  APP_BANNER_DISMISSED = 'App Banner Dismissed',
+}
+
+export interface AppStoreBannerProps {
+  CloseButton?({ onClose }: DefaultCloseButtonProps): React.ReactNode;
+}
+
+export function AppStoreBanner({
+  CloseButton = DefaultCloseButton,
+}: AppStoreBannerProps) {
+  const { appStoreUrl, closeBanner } = useAppStoreBannerState();
+  const { t } = useTranslation();
+  return (
+    appStoreUrl && (
+      <div
+        role="banner"
+        aria-live="polite"
+        className="sticky flex h-14 min-h-14 w-full items-center border-b-[1px] border-solid border-neutral-300 bg-neutral-100 px-3"
+      >
+        <CloseButton onClose={closeBanner} />
+        <div className="flex flex-grow gap-2">
+          <div className="items-center justify-center self-center rounded-md border-[1px] border-neutral-200 bg-white">
+            <EmapsIcon size={40} />
+          </div>
+          <div className="content-center font-inter text-neutral-600">
+            <div className="text-sm font-semibold">Electricity Maps</div>
+            <div className="text-xs">{t('app-banner.description')}</div>
+          </div>
+        </div>
+        <Button
+          size="md"
+          backgroundClasses={'h-9'}
+          href={appStoreUrl}
+          onClick={trackCTAClick}
+        >
+          {t('app-banner.cta')}
+        </Button>
+      </div>
+    )
+  );
+}
+
+export const trackCTAClick = () => trackEvent(TrackEvents.APP_BANNER_CTA_CLICKED);
+
+interface DefaultCloseButtonProps {
+  onClose(): void;
+}
+
+function DefaultCloseButton({ onClose }: DefaultCloseButtonProps) {
+  const { t } = useTranslation();
+  return (
+    <button
+      aria-label={t('misc.dismiss')}
+      data-testid="dismiss-btn"
+      onClick={onClose}
+      className="pointer-events-auto flex h-6 w-6 items-center justify-center self-center pr-2 text-neutral-400"
+    >
+      <X />
+    </button>
+  );
+}
+
+interface AppStoreBannerState {
+  appStoreUrl: string | undefined;
+  closeBanner(): void;
+}
+
+const useAppStoreBannerState = (): AppStoreBannerState => {
+  const [appStoreUrl, setAppStoreUrl] = useState(getAppStoreUrl());
+
+  return {
+    appStoreUrl,
+    closeBanner: () => {
+      localStorage.setItem(IS_APP_BANNER_DISMISSED, 'true');
```

Comment:
Yeah, I wasn't sure what would be best practice here (see [this convo](https://linear.app/electricitymaps/project/app-convert-mobile-web-users-to-app-users-463e8430b370#comment-12c5cc6b)). I see that atomWithStorage makes use of local storage, but we don't actually need this piece of state anywhere else in the app. Happy to switch it over to `atomWithStorage` if that's what you recommend, though!

Reply:
I think it would just be nicer to have as much as possible in atoms to try and keep the codebase clean, we have other atoms that are single use as well.